Updates to EF Core 5.0 plan (#2452)
dotnet/EntityFramework.Docs: entity-framework/core/what-is-new/ef-core-5.0/plan.md

It is expected that there will be some breaking changes for database providers, especially around TPT support. However, we expect the work to update a provider for 5.0 will be less than was required to update for 3.0.

## Themes

We have extracted a few major areas or themes which will form the basis for the large investments in EF Core 5.0.

## Fully transparent many-to-many mapping by convention

Lead developers: @smitpatel, @AndriySvyryd, and @lajones

Tracked by [#10508](https://github.com/aspnet/EntityFrameworkCore/issues/10508)

T-shirt size: L

Status: In-progress

Many-to-many is the [most requested feature](https://github.com/aspnet/EntityFrameworkCore/issues/1368) (~506 votes) on the GitHub backlog.

Support for many-to-many relationships can be broken down into three major areas:

* Skip navigation properties--covered by the next theme.
* Property-bag entity types. These allow a standard CLR type (e.g. `Dictionary`) to be used for entity instances such that an explicit CLR type is not needed for each entity type. Tracked by [#9914](https://github.com/aspnet/EntityFrameworkCore/issues/9914).
* Sugar for easy configuration of many-to-many relationships.

> In addition to the skip navigation supprt, we are now pulling these other areas of many-to-many into EF Core 5.0 so as to provide a complete experience.

Reviewer: ```suggestion
In addition to the skip navigation support, we are now pulling these other areas of many-to-many into EF Core 5.0 so as to provide a complete experience.
```

Status: In-progress

Filtered Include is a highly-requested feature (~376 votes; 2nd overall) that isn't a huge amount of work, and that we believe will unblock or make easier many scenarios that currently require model-level filters or more complex queries.

## Split Include

Lead developer: @smitpatel

Tracked by [#20892](https://github.com/dotnet/efcore/issues/20892)

T-shirt size: L

Status: In-progress

EF Core 3.0 changed the default behavior to create a single SQL query for a given LINQ query.

> This caused large performance regressions for queries that use Include for multiple collections.

Reviewer: ```suggestion
This caused large performance regressions for queries that uses Include for multiple collections.
```

In EF Core 5.0, we are retaining the new default behavior.

> However, EF Core 5.0 will now allow generation of multiple Includes where having a single query is not needed. 

Reviewer: ```suggestion
However, EF Core 5.0 will now allow generation of multiple queries for collection Includes where having a single query is causing bad performance.. 
```
